### PR TITLE
Feature/logger share button

### PR DESCRIPTION
**Pull request by bernhard-eiling**

### CODE REVIEW

**Comment by bernhard-eiling** [comment]
We have the option to show the share button in the navigation bar but then we would lose the share functionality when using the logger as a standalone plugin.
@brototyp @thorstenstark Your thoughts to this issue?

**Comment by thorstenstark** [comment]
I would write a note which states that the use of the share functionality requires the loggerViewController to be inside of a NavigationController.

**Comment by bernhard-eiling** [comment]
I'm tending to keep the modularity of the DeveloperOverlay while keeping it's full functionality.
@brototyp What is your opinion about this?

**Comment by brototyp** [comment]
I would say it is fine this way. We don't lose the core-functionality of the LoggerOverlay. 
There are two things i would add:
- rename `- (IBAction)shareButtonTapped` to `- (IBAction)shareButtonTapped:(id)sender`
- make this method public

File: TBODeveloperOverlay/TBODeveloperOverlay/TBOModalNavigationController.m [hunk 1]
```diff
@@ -18,6 +18,9 @@ @implementation TBOModalNavigationController
 
 - (void)pushViewController:(UIViewController *)viewController animated:(BOOL)animated {
     [super pushViewController:viewController animated:animated];
+    if ([viewController respondsToSelector:@selector(shareButtonTapped:)]) {
+        viewController.navigationItem.rightBarButtonItem = [[UIBarButtonItem alloc] initWithBarButtonSystemItem:UIBarButtonSystemItemAction target:viewController action:@selector(shareButtonTapped:)];
+    }
```
Comment (brototyp) [inline TBODeveloperOverlay/TBODeveloperOverlay/TBOModalNavigationController.m]: I would rather let that do the ViewController itself.

Comment (bernhard-eiling) [inline TBODeveloperOverlay/TBODeveloperOverlay/TBOModalNavigationController.m]: Better.

**Pull request merged**